mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -626,10 +621,25 @@ def init_resultsinfo(self):
         # )
         self.control = SimControl(max_steps=self.TOTAL_TIMESTEPS)
 
-    def init_warpx(self):
-        sim = picmi.Simulation(
+    def init_simulation(self, **kw):
+        warpx_collisions = kw.pop('warpx_collisions')
```

Comment:
`warpx_collisions` should be what's created in the `inert_gas_ionization` function above. So that should be integrated into this rather than being a passed kw. If that function isn't run, defaulting warpx_collisions to None should be fine.